internal/provider/fwprovider/auth_kerberos.go

```diff
@@ -0,0 +1,91 @@
+package fwprovider
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework-validators/boolvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/provider/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+)
+
+func AuthLoginKerberosSchema() schema.Block {
+	return mustAddLoginSchema(&schema.ListNestedBlock{
+		Description: "Login to vault using the kerberos method",
+		NestedObject: schema.NestedBlockObject{
+			Attributes: map[string]schema.Attribute{
+				consts.FieldToken: schema.StringAttribute{
+					Optional:    true,
+					Description: "Simple and Protected GSSAPI Negotiation Mechanism (SPNEGO) token",
+					// ValidateFunc: validateKRBNegToken,
+				},
+				consts.FieldUsername: schema.StringAttribute{
+					Optional:    true,
+					Description: "The username to login into Kerberos with.",
+					Validators: []validator.String{
+						stringvalidator.ConflictsWith(
+							path.MatchRelative().AtName(consts.FieldToken),
+						),
+					},
+				},
+				consts.FieldService: schema.StringAttribute{
+					Optional:    true,
+					Description: "The service principle name.",
+					Validators: []validator.String{
+						stringvalidator.ConflictsWith(
+							path.MatchRelative().AtName(consts.FieldToken),
+						),
+					},
+				},
+				consts.FieldRealm: schema.StringAttribute{
+					Optional:    true,
+					Description: "The Kerberos server's authoritative authentication domain",
+					Validators: []validator.String{
+						stringvalidator.ConflictsWith(
+							path.MatchRelative().AtName(consts.FieldToken),
+						),
+					},
+				},
+				consts.FieldKRB5ConfPath: schema.StringAttribute{
+					Optional:    true,
+					Description: "A valid Kerberos configuration file e.g. /etc/krb5.conf.",
+					// ValidateFunc:  validateFileExists,
+					Validators: []validator.String{
+						stringvalidator.ConflictsWith(
+							path.MatchRelative().AtName(consts.FieldToken),
+						),
+					},
+				},
+				consts.FieldKeytabPath: schema.StringAttribute{
+					Optional:    true,
+					Description: "The Kerberos keytab file containing the entry of the login entity.",
+					// ValidateFunc:  validateFileExists,
```

Comment:
similar question as above for validating if the file exists

Reply:
done in https://github.com/hashicorp/terraform-provider-vault/pull/2090/commits/3fbf25a94b2bb323b9a4a3abf34fdb5c1cb1112f